### ISSUE TITLE
[question] cross-compilation issues for packages relying on generators built during compilation

### ISSUE THREAD

**Issue by bram-vanderstreeck** [conan-io/conan#11380]
For one of our projects we are using conan as our package manager. We are using conan in CMake through the CMake wrapper (https://github.com/conan-io/cmake-conan). Our project is intended to run on multiple platforms, so in order to accomplish this we've built a number of our own toolchains through crosstool-ng (https://crosstool-ng.github.io) and set up the corresponding conan profiles and CMake toolchain files for cross compilation.

Because of our pipeline needs, we are building everything in docker images, with our build image being based on Ubuntu 20.04. Some packages seem to be cross-compiling perfectly fine, e.g. https://conan.io/center/redis-plus-plus, https://conan.io/center/protobuf, https://conan.io/center/asio, ... and are perfectly usable. However certain packages seem to be giving us issues. Especially those which seem to rely on generated binaries used in turn as generators. One example of such a package is https://conan.io/center/sdbus-cpp and more in particular it's dependency selinux, which fails with:

```
libselinux/3.3: Applying build-requirement: flex/2.6.4
libselinux/3.3: Applying build-requirement: m4/1.4.19
libselinux/3.3: Configuring sources in /home/bramv/.conan/data/libselinux/3.3/_/_/source

libselinux/3.3: Copying sources to build folder
libselinux/3.3: Building your package in /home/bramv/.conan/data/libselinux/3.3/_/_/build/96bcd54f9b12eb32134c70b26750be8e77157885
libselinux/3.3: Generator txt created conanbuildinfo.txt
libselinux/3.3: Aggregating env generators
libselinux/3.3: Calling build()
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o assertion.o assertion.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o avrule_block.o avrule_block.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o avtab.o avtab.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o boolean_record.o boolean_record.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o booleans.o booleans.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o conditional.o conditional.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o constraint.o constraint.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o context.o context.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o context_record.o context_record.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o debug.o debug.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ebitmap.o ebitmap.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o expand.o expand.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o handle.o handle.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o hashtab.o hashtab.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o hierarchy.o hierarchy.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ibendport_record.o ibendport_record.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ibendports.o ibendports.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ibpkey_record.o ibpkey_record.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ibpkeys.o ibpkeys.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o iface_record.o iface_record.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o interfaces.o interfaces.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o kernel_to_cil.o kernel_to_cil.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o kernel_to_common.o kernel_to_common.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o kernel_to_conf.o kernel_to_conf.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o link.o link.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o mls.o mls.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o module.o module.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o module_to_cil.o module_to_cil.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o node_record.o node_record.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o nodes.o nodes.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o optimize.o optimize.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o polcaps.o polcaps.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o policydb.o policydb.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o policydb_convert.o policydb_convert.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o policydb_public.o policydb_public.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o policydb_validate.o policydb_validate.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o port_record.o port_record.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ports.o ports.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o services.o services.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o sidtab.o sidtab.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o symtab.o symtab.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o user_record.o user_record.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o users.o users.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o util.o util.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o write.o write.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ../cil/src/cil.o ../cil/src/cil.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ../cil/src/cil_binary.o ../cil/src/cil_binary.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ../cil/src/cil_build_ast.o ../cil/src/cil_build_ast.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ../cil/src/cil_copy_ast.o ../cil/src/cil_copy_ast.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ../cil/src/cil_find.o ../cil/src/cil_find.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ../cil/src/cil_fqn.o ../cil/src/cil_fqn.c
flex -o ../cil/src/cil_lexer.c ../cil/src/cil_lexer.l
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ../cil/src/cil_list.o ../cil/src/cil_list.c
aarch64-cortex_a53-linux-gnu-gcc -g -fPIC -I. -I../include -D_GNU_SOURCE -I../cil/include -fPIC -c -o ../cil/src/cil_log.o ../cil/src/cil_log.c
/home/bramv/.conan/data/flex/2.6.4/_/_/package/1f477ddc5d737c4319f1655902f1498a01405d90/bin/flex: 1: Syntax error: Unterminated quoted string
make: *** [Makefile:71: ../cil/src/cil_lexer.c] Error 2
make: *** Waiting for unfinished jobs....
libselinux/3.3: 
WARN: Remotes registry file missing, creating default one in /home/bramv/.conan/remotes.json
libselinux/3.3: ERROR: Package '96bcd54f9b12eb32134c70b26750be8e77157885' build failed
libselinux/3.3: WARN: Build folder /home/bramv/.conan/data/libselinux/3.3/_/_/build/96bcd54f9b12eb32134c70b26750be8e77157885
ERROR: libselinux/3.3: Error in build() method, line 79
        env_build.make(args=args)
        ConanException: Error 2 while executing make 'libsepol.a' -j16
CMake Error at cmake/conan.cmake:638 (message):
  Conan install failed='1'
Call Stack (most recent call first):
  service/CMakeLists.txt:21 (conan_cmake_install)


-- Configuring incomplete, errors occurred!
See also "/home/bramv/build/build_armv8_64_deb/CMakeFiles/CMakeOutput.log".
The terminal process "/usr/bin/bash '-c', 'cmake -S . -B build_armv8_64_deb -G Ninja -D CMAKE_TOOLCHAIN_FILE=cmake/toolchains/aarch64-cortex_a53-linux-gnu.cmake -D BUILD_SHARED_LIBS=ON -D CMAKE_BUILD_TYPE=Debug -D STAGE_SUB_FOLDER=stage_armv8_64_deb -D CMAKE_PREFIX_PATH="/home/bramv/build/dependencies_armv8_64" -D CMAKE_MODULE_PATH=/home/bramv/build/stage_armv8_64_deb && ninja -C build_armv8_64_deb'" terminated with exit code: 1.
```

As can be seen in the output above, the execution of `/home/bramv/.conan/data/flex/2.6.4/_/_/package/1f477ddc5d737c4319f1655902f1498a01405d90/bin/flex` seems to be causing the issue. It is my feeling that the error message `Syntax error: Unterminated quoted string` is not the actual issue occurring here, considering the following:

```
bramv@iprc00371:~/build$ uname -a
Linux iprc00371 5.14.0-1038-oem #42-Ubuntu SMP Thu May 19 05:03:08 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
bramv@iprc00371:~/build$ /home/bramv/.conan/data/flex/2.6.4/_/_/package/1f477ddc5d737c4319f1655902f1498a01405d90/bin/flex
bash: /home/bramv/.conan/data/flex/2.6.4/_/_/package/1f477ddc5d737c4319f1655902f1498a01405d90/bin/flex: cannot execute binary file: Exec format error
bramv@iprc00371:~/build$ readelf -h /home/bramv/.conan/data/flex/2.6.4/_/_/package/1f477ddc5d737c4319f1655902f1498a01405d90/bin/flex
ELF Header:
  Magic:   7f 45 4c 46 02 01 01 00 00 00 00 00 00 00 00 00 
  Class:                             ELF64
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              EXEC (Executable file)
  Machine:                           AArch64
  Version:                           0x1
  Entry point address:               0x401bc0
  Start of program headers:          64 (bytes into file)
  Start of section headers:          553568 (bytes into file)
  Flags:                             0x0
  Size of this header:               64 (bytes)
  Size of program headers:           56 (bytes)
  Number of program headers:         9
  Size of section headers:           64 (bytes)
  Number of section headers:         36
  Section header string table index: 35
```

It looks more like the build fails because it can't execute the flex binary since the host and build platforms are not the same.

The conan profile file for this particular toolchain is the following (we're using a jinja template):

```
toolchain_root={{os.getenv("CI_PROJECT_DIR")}}
toolchain_name=aarch64-cortex_a53-linux-gnu
toolchain=$toolchain_root/$toolchain_name

[build_requires]

[conf]
tools.gnu.make:jobs={{os.cpu_count()}}
tools.ninja:jobs={{os.cpu_count()}}

[env]
CONAN_CPU_COUNT={{os.cpu_count()}}
CONAN_CMAKE_FIND_ROOT_PATH=$toolchain/bin/
CONAN_CMAKE_SYSROOT=$toolchain/$toolchain_name/sysroot/
CONAN_CMAKE_GENERATOR=Ninja
PATH=[$toolchain/bin/]
CHOST=$toolchain_name
CC=$toolchain_name-gcc
CXX=$toolchain_name-g++
AR=$toolchain_name-ar
AS=$toolchain_name-as
GDB=$toolchain_name-gdb
LD=$toolchain_name-ld
NM=$toolchain_name-nm
OBJCOPY=$toolchain_name-objcopy
OBJDUMP=$toolchain_name-objdump
RANLIB=$toolchain_name-ranlib
STRIP=$toolchain_name-strip

[options]

[settings]
os=Linux
arch=armv8
compiler=gcc
compiler.version=11.2
compiler.libcxx=libstdc++11
build_type={{os.getenv("CMAKE_BUILD_TYPE")}}
```

While the contents of the CMake toolchain file is:

```
###################################################################################################
## Toolchain options (allows overriding locations for local build)
###################################################################################################

include("${CMAKE_CURRENT_LIST_DIR}/../celp/toolchain_config.cmake")

###################################################################################################
## Toolchain properties
###################################################################################################

set(toolchain_name "aarch64-cortex_a53-linux-gnu")
set(toolchain_root "${TOOLCHAIN_FOLDER}/${toolchain_name}")
set(stage_root "${STAGE_FOLDER}/${STAGE_SUB_FOLDER}")

set(CMAKE_SYSTEM_NAME Linux)
set(CMAKE_SYSTEM_PROCESSOR armv8)
set(CMAKE_SYSROOT ${toolchain_root}/${toolchain_name}/sysroot)

set(CMAKE_STAGING_PREFIX ${stage_root})

set(CMAKE_C_COMPILER ${toolchain_root}/bin/${toolchain_name}-gcc)
set(CMAKE_CXX_COMPILER ${toolchain_root}/bin/${toolchain_name}-g++)
set(CMAKE_AR ${toolchain_root}/bin/${toolchain_name}-ar)
set(CMAKE_OBJCOPY ${toolchain_root}/bin/${toolchain_name}-objcopy)
set(CMAKE_OBJDUMP ${toolchain_root}/bin/${toolchain_name}-objdump)
set(CMAKE_RANLIB ${toolchain_root}/bin/${toolchain_name}-ranlib)
set(CMAKE_STRIP ${toolchain_root}/bin/${toolchain_name}-strip)

set(CMAKE_BUILD_RPATH "/opt/${toolchain_name}/lib64")
set(CMAKE_INSTALL_RPATH "/opt/${toolchain_name}/lib64")
set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-dynamic-linker=/opt/${toolchain_name}/lib64/ld-linux-aarch64.so.1")
set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-dynamic-linker=/opt/${toolchain_name}/lib64/ld-linux-aarch64.so.1")

set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE BOTH)
```

The same packages compile just fine for x86_64 with the `Debug` or `Release` build type, but not for the `RelWithDebInfo` build type which seems to yield the following error:

```
pkgconf/1.7.4: Applying build-requirement: meson/0.62.1
pkgconf/1.7.4: Applying build-requirement: ninja/1.10.2
Downloading conan_sources.tgz
pkgconf/1.7.4: Configuring sources in /home/bramv/.conan/data/pkgconf/1.7.4/_/_/source

pkgconf/1.7.4: Copying sources to build folder
pkgconf/1.7.4: Building your package in /home/bramv/.conan/data/pkgconf/1.7.4/_/_/build/252f7e09f641bef29a75fc98d05ac217b842cde4
pkgconf/1.7.4: Generator txt created conanbuildinfo.txt
pkgconf/1.7.4: Aggregating env generators
pkgconf/1.7.4: Calling build()
The Meson build system
Version: 0.62.1
Source dir: /home/bramv/.conan/data/pkgconf/1.7.4/_/_/build/252f7e09f641bef29a75fc98d05ac217b842cde4/source_subfolder
Build dir: /home/bramv/.conan/data/pkgconf/1.7.4/_/_/build/252f7e09f641bef29a75fc98d05ac217b842cde4/build_subfolder
Build type: native build
Project name: pkgconf
Project version: 1.7.4
C compiler for the host machine: x86_64-ubuntu20.04-linux-gnu-gcc (gcc 11.2.0 "x86_64-ubuntu20.04-linux-gnu-gcc (crosstool-NG 1.24.0.535_74d6f26) 11.2.0")
C linker for the host machine: x86_64-ubuntu20.04-linux-gnu-gcc ld.bfd 1.24.0.535
Host machine cpu family: x86_64
Host machine cpu: x86_64
Configuring config.h using configuration
Build targets in project: 2
NOTICE: Future-deprecated features used:
 * 0.56.0: {'meson.build_root', 'meson.source_root'}

pkgconf 1.7.4

  User defined options
    backend        : ninja
    bindir         : bin
    buildtype      : debugoptimized
    default_library: static
    includedir     : include
    libdir         : lib
    libexecdir     : bin
    prefix         : /home/bramv/.conan/data/pkgconf/1.7.4/_/_/package/252f7e09f641bef29a75fc98d05ac217b842cde4
    sbindir        : bin
    sharedstatedir : /home/bramv/.conan/data/pkgconf/1.7.4/_/_/package/252f7e09f641bef29a75fc98d05ac217b842cde4/bin/share
    tests          : False


ERROR: Could not detect Ninja v1.8.2 or newer

A full log can be found at /home/bramv/.conan/data/pkgconf/1.7.4/_/_/build/252f7e09f641bef29a75fc98d05ac217b842cde4/build_subfolder/meson-logs/meson-log.txt
pkgconf/1.7.4: 
pkgconf/1.7.4: ERROR: Package '252f7e09f641bef29a75fc98d05ac217b842cde4' build failed
pkgconf/1.7.4: WARN: Build folder /home/bramv/.conan/data/pkgconf/1.7.4/_/_/build/252f7e09f641bef29a75fc98d05ac217b842cde4
ERROR: pkgconf/1.7.4: Error in build() method, line 101
        meson = self._configure_meson()
while calling '_configure_meson', line 84
        meson.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
        ConanException: Error 1 while executing meson "/home/bramv/.conan/data/pkgconf/1.7.4/_/_/build/252f7e09f641bef29a75fc98d05ac217b842cde4/source_subfolder" "/home/bramv/.conan/data/pkgconf/1.7.4/_/_/build/252f7e09f641bef29a75fc98d05ac217b842cde4/build_subfolder" --backend=ninja -Dprefix="/home/bramv/.conan/data/pkgconf/1.7.4/_/_/package/252f7e09f641bef29a75fc98d05ac217b842cde4" -Dlibdir="lib" -Dbindir="bin" -Dsbindir="bin" -Dlibexecdir="bin" -Dincludedir="include" -Ddefault_library="static" -Dtests="False" -Dsharedstatedir="/home/bramv/.conan/data/pkgconf/1.7.4/_/_/package/252f7e09f641bef29a75fc98d05ac217b842cde4/bin/share" --buildtype=debugoptimized
CMake Error at cmake/conan.cmake:638 (message):
  Conan install failed='1'
Call Stack (most recent call first):
  service/CMakeLists.txt:21 (conan_cmake_install)


-- Configuring incomplete, errors occurred!
See also "/home/bramv/build/build_x86_64_rel/CMakeFiles/CMakeOutput.log".
The terminal process "/usr/bin/bash '-c', 'cmake -S . -B build_x86_64_rel -G Ninja -D CMAKE_TOOLCHAIN_FILE=cmake/toolchains/x86_64-ubuntu20.04-linux-gnu.cmake -D ENABLE_UNIT_TESTS=ON -D BUILD_SHARED_LIBS=ON -D CMAKE_BUILD_TYPE=Release -D STAGE_SUB_FOLDER=stage_x86_64_rel -D CMAKE_PREFIX_PATH="/home/bramv/build/dependencies_x86_64" -D CMAKE_MODULE_PATH=/home/bramv/build/stage_x86_64_rel && ninja -C build_x86_64_rel'" terminated with exit code: 1.
```

In both cases I feel like I'm missing something obvious here and was wondering if maybe you could shed a light on my erroneous ways. Any advice would be appreciated and I'll gladly provide more info if required.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by curlybeast** [comment]
This issue is affecting me too. Did you find any work around for this yet?

**Comment by memsharded** [comment]
Hi @curlybeast 

Sorry @bram-vanderstreeck this was not managed back then, we have a ton of activity around our repos, so we basically cannot process all notifications, and eventually some tickets might fall through the cracks. Feel free to ping or comment on issues to resurface them.

As this issue is a bit old, I'd suggest the following approach:

- It is super difficult to support when there are many moving pieces. I'd recommend to remove the ``cmake-conan`` integration from this one, as it is probably unrelated, and try to reproduce without it, directly from ``conan install`` command line.
- Same for Meson, so the above is reporting a problem with Meson build that can't find Ninja, this is probably because of missing the 2 build and host profiles (see below).
- Same for ConanCenter recipes, some of them can be very complex, so in order to evaluate if it is a Conan issue or a recipe issue, it is very recommended to reduce the problem to simple recipes, like those that can be produced with ``conan new mypkg/0.1 -m=cmake_lib`` or ``conan new myapp/0.1 -m=cmake_exe``, and then add minor changes to them
- For cross-building it is very important to use the 2 build&host profiles always, it is very possible that recipes in ConanCenter can only cross-compile correctly if using the 2 profiles (this 2 profiles mode is already default in Conan 2.0, but needs to be explicitly opted-in with ``-pr:b=mybuildprofile`` in 1.X)
- It is also important to modernize the tools, like using ``CMakeDeps`` and ``CMakeToolchain`` and not ``cmake`` or ``cmake_find_package`` generators.
- When reporting please provide more details, like the actual command line used to fire the behavior, and the full logs of that command. Also possible ``conanfile.py`` being used, as well as the exact Conan version (Conan 2.0 is out, it is more likely to get prioritized if reported against 2.0).

To summarize, the best would be something as small as possible (minimum reproducible example) that is very focused on one thing, and that we could reproduce on our end, that really helps to investigate things and fix issue quickly, or at least being able to help faster.

**Comment by AbrilRBS** [comment]
This went inactive twice now — as memsharded said in-thread, it fell through the cracks the first time, and a second independent reporter (@curlybeast) hit the same problem without it moving forward. If the crosstool-ng cross-compilation issue is still live for you on Conan 2, let us know here; that's what helps us prioritize it.
